src/metabase/models/field_values.clj

```diff
@@ -334,14 +334,33 @@
       (log/error e (trs "Error fetching field values"))
       nil)))
 
+(defn- report-and-fix-duplicates! [rows]
+  (if (<= (count rows) 1)
+    (first rows)
+    ;; todo - rather use a descending comparator
+    (let [[latest & duplicates] (reverse (sort-by :updated-at rows))]
+      ;; send telemetry to snowplow
+      (t2/delete! FieldValues :id [:in (map :id duplicates)])
+      latest)))
+
+(defn- get-field-values-for-field
```

Comment:
This function name should reflect that it only selects one field values instance. How about `select-one-field-values`?

Reply:
How about `select-latest-field-values`?

Reply:
I'm a bit torn about this, as needing to pick from multiple rows is an anomaly and I wanted to hide that detail away.

... or maybe I should ilt in the other direction and mention the inner hack in the name, maybe something like `get-latest-field-values-and-delete-stale-rows!` 🤔

Reply:
better!